### DISCUSSION TITLE
Quickest way to upgrade from Google Billing 3 to 4?

### DISCUSSION THREAD

**Discussion by the-dut** [hyochan/react-native-iap#2037] [General]
Hi.  Somehow I missed the need to migrate from Google Billing 3.0 and am looking for the quickest, most non-intrusive, way to get to V4 or later.  Was hoping to do so without a ton of changes to my app.  What did you guys do?  And where there a lot of breaking changes to your apps?

**Comment by andresesfm** [comment]
v 10.0.0 and up will have the Google Play Billing library 5.0 but it will require several changes. Not sure if there's an easy option since the native API changed substantially.

**Comment by SpaceSimon** [comment]
I also need to upgrade to Billing 4 and would like to know which RNIAP version does this. If some people have any advice or encountered specific problems while doing so, I'll take that too :)

Thanks

**Comment by SpaceSimon** [comment]
OK so here is what I did:
- I updated to 6.2.0 and encountered a problem, that I solved adding `missingDimensionStrategy 'store', 'play'` in android/app/build.gradle
- I updated to 7.0.0, which uses Billing 4. No problem encountered.
- I updated to 7.2.0 and encountered an error saying 'E_IAP_NOT_AVAILABLE'.
- I updated to 7.2.1 and it seemed it solved this problem.
- I updated to 7.5.6 without any problem, but encountered a problem when uploading to Play Store: I had to change `targetSdkVersion` and `compileSdkVersion` so they target 31, and had to add `android:exported="true"` in every `<activity>` of AndroidManifest.xml
- I had problems trying to update to 8.0.0 so for now I'm staying in 7.5.6.
